lib/rules/best-practises/no-empty-blocks.js

```diff
@@ -27,7 +27,15 @@ class NoEmptyBlocksChecker extends BaseChecker {
     this._validateContractPartsCount(node)
   }
 
+  FunctionDefinition(node) {
+    node.body.shouldIgnore = node.isConstructor && node.modifiers.length > 0
```

Comment:
Keep in mind the AST is shared for all visitors, so this property will exist in the Blocks that are bodies of a function in all rule definitions that are executed after this one.

Therefore, an effort should be made not to pollute the AST with information not globally relevant.

My personally preferred approach would be to add the `isConstructor` and `modifiers` fields of the `FunctionDefinition` node to its `body`, and then assert `isConstructors && modifiers.length > 0` in the listener for `Block` nodes.

Having those two names in the `Block` node makes it more clear what they are if they show up in the execution of another rule.

However, I'm not sure that it's a good practise to modify the AST that's gonna be used by other rules, and I'd like the opinion of @fvictorio on this topic. (I grepped a bit and couldn't find many usages of the `node` variable as an lvalue)

If it's problematic to modify the AST in this way, then you'd have to store the information of what blocks to ignore in the `ComplexityScore` class. One such way could be to have an object whose keys are a string that looks like ``${range[0]-range[1]}``, and it's value is wether to ignore the block or not.

Reply:
The way this is done in other rules is by adding a field to the checker instance. So:

```
this.shouldIgnore = ...
```

Agree that modifying the node is not a good idea.

Reply:
@fvictorio @juanpcapurro I agree what you've mentioned. Howeve whether to be constructor could be validated in `FunctionDefinition`. Therefore, I needed to recognize if its block is passed by constructor or not. Do you have any other good idea to pass it to the block.

Reply:
@fvictorio Thank for good to me. I've change it to use node.parent and checked it working properly.